crates/iota-metrics/src/hardware_metrics.rs

```diff
@@ -0,0 +1,280 @@
+                .const_label("disk_total_bytes", disk_total_bytes.to_string())
+                .const_label("disk_total_space_human", human_fmt_bytes(disk_total_bytes)),
+        )
+        .map_err(HardwareMetricsErr::ErrCreateMetric)?;
+        disk_collector.set(disk_total_bytes as i64);
+
+        Ok(Self {
+            cpu_model: cpu_model.to_string(),
+            cpu_vendor_id: cpu_vendor_id.to_string(),
+            cpu_core_count,
+            memory_collector,
+            disk_collector,
+            is_docker: is_running_in_docker(),
+        })
+    }
+
+    fn label(name: &str, value: impl ToString) -> LabelPair {
+        let mut label = LabelPair::new();
+        label.set_name(name.to_string());
+        label.set_value(value.to_string());
+        label
+    }
+
+    fn cpu_metric(&self) -> Metric {
+        let mut metric = Metric::new();
+        metric.set_label({
+            vec![
+                Self::label("cpu_model", &self.cpu_model),
+                Self::label("cpu_vendor_id", &self.cpu_vendor_id),
+                Self::label(
+                    "cpu_core_count",
+                    self.cpu_core_count.map_or_else(
+                        || "cpu_core_count_unavailable".to_string(),
+                        |c| c.to_string(),
+                    ),
+                ),
+                Self::label("cpu_arch", System::cpu_arch()),
+            ]
+            .into()
+        });
+        metric
+    }
+    fn cpu_metric_family(&self) -> MetricFamily {
+        let mut mf = MetricFamily::new();
+        mf.set_name("cpu_specs".to_owned());
+        mf.set_help("CPU specs (brand,vendor,cores)".to_owned());
+        mf.set_field_type(prometheus::proto::MetricType::COUNTER);
+        mf.set_metric(vec![self.cpu_metric()].into());
+        mf
+    }
+
+    fn system_metric(&self) -> Metric {
+        let mut metric = Metric::new();
+        metric.set_label({
+            vec![
+                Self::label("is_docker", self.is_docker.to_string()),
+                Self::label(
+                    "os_version",
+                    System::long_os_version()
+                        .unwrap_or_else(|| "os_version_unavailable".to_string()),
+                ),
+            ]
+            .into()
+        });
+        metric
+    }
+    fn system_metric_family(&self) -> MetricFamily {
+        let mut mf = MetricFamily::new();
+        mf.set_name("system_info".to_owned());
+        mf.set_help("System info (OS, version, is_docker, ...)".to_owned());
+        mf.set_field_type(prometheus::proto::MetricType::COUNTER);
+        mf.set_metric(vec![self.system_metric()].into());
+        mf
+    }
+}
+
+impl Collector for HardwareSpecs {
+    fn desc(&self) -> Vec<&Desc> {
+        let mut descs = Vec::new();
+        descs.extend(self.memory_collector.desc());
+        descs.extend(self.disk_collector.desc());
+        descs
+    }
+
+    fn collect(&self) -> Vec<prometheus::proto::MetricFamily> {
```

Comment:
```suggestion
    fn collect(&self) -> Vec<MetricFamily> {
```

Reply:
changed already in the latest push, but good point